### ISSUE TITLE
Impossible to create multiple instances of MakePaymentModule class

### ISSUE THREAD

**Issue by AwesomeAndrewMK** [google-pay/react-native-make-payment#3]
When following the instructions for working with Google Pay, you'll need to initialize a payment request instance as follows:
`const paymentRequest = new PaymentRequest(paymentMethods, paymentDetails);`
which includes payment details which are unique for each payment.
This initialization includes payment details that are unique to each payment transaction. However, when attempting to create a new instance with updated payment details, you may encounter the following issue:

```
**Possible Unhandled Promise Rejection (id: 0):
Error: You attempted to set the key `softwareInfo` with the value `{"id":"react-native-make-payment","version":"0.1.0"}` on an object that is meant to be immutable and has been frozen.
Error: You attempted to set the key `softwareInfo` with the value `{"id":"react-native-make-payment","version":"0.1.0"}` on an object that is meant to be immutable and has been frozen.**
```
<img width="631" alt="Screenshot 2023-09-21 at 16 31 17" src="https://github.com/google-pay/react-native-make-payment/assets/42909164/a0c51524-ffc7-483a-afda-afbb9f3f4329">

I kindly request your assistance in resolving this issue or providing instructions on how to bypass it.

**Comment by AwesomeAndrewMK** [comment]
@stephenmcd, @JlUgia Could you have a look at this, please? Or is it better to raise this ticket somewhere else?

**Comment by stephenmcd** [comment]
Thanks for the report, this is definitely the correct place for it. We'll update here once we have a resolution.

**Comment by stephenmcd** [comment]
Hi, I wasn't able to reproduce the error, although there's no doubt you're somehow getting it. 

You'll need to provide more info (preferably some code to try) showing how you're creating and reusing the `paymentMethods` arg in `PaymentRequest(paymentMethods, paymentDetails)`. 

Here's what I tried by editing the included demo app, but it did not produce the error:

```
// this line is in the demo app
const paymentRequest = new PaymentRequest(paymentMethods, paymentDetails);

// added these, but no error
const paymentMethodsSameObj = [paymentMethods[0]];
const paymentRequestObj = new PaymentRequest(paymentMethodsSameObj, paymentDetails);
const paymentMethodsNewObj = [Object.assign({}, paymentMethods[0])];
const paymentRequestNewObj = new PaymentRequest(paymentMethodsNewObj, paymentDetails);
```

What you're doing must differ from this.

**Comment by AwesomeAndrewMK** [comment]
Hi @stephenmcd. Thanks for the prompt response!

In the scenario you mentioned, everything is fine, but please check this commit with my changes and comments, which will help you reproduce the issue: [https://github.com/google-pay/react-native-make-payment/pull/4/commits/9120b25d52d6cf044e9b0a622a8b0b1f709a8b95](https://github.com/google-pay/react-native-make-payment/pull/4/commits/9120b25d52d6cf044e9b0a622a8b0b1f709a8b95)

Just for a demo, I've replaced the image link, as the previous one wasn't working.

**Comment by stephenmcd** [comment]
Thank you so much for the code sample. I was able to reproduce the error and I just released version 0.1.1 to NPM with the fix: e1a35f0fef03b0e589e6a676ed56d5f321790644
